docs/resources/entitlements.md

```diff
@@ -0,0 +1,88 @@
+---
+subcategory: "Security"
+---
+# databricks_entitlements Resource
+
+This resource allows you to set entitlements to existing [databricks_users](user.md), [databricks_group](group.md) or [databricks_service_principal](service_principal.md)
+
+## Example Usage
+
+Setting entitlements for a regular user:
+
+```hcl
+data "databricks_user" "me" {
+  user_name = "me@example.com"
+}
+
+resource "databricks_entitlements" "me" {
+  user_id                    = data.databricks_user.me.id
+  allow_cluster_create       = true
+  allow_instance_pool_create = true
+}
+```
+
+Setting entitlements for a service principal:
+
+```hcl
+data "databricks_service_principal" "service_principal" {
+  application_id = "11111111-2222-3333-4444-555666777888"
+}
+
+resource "databricks_entitlements" "service_principal" {
+  service_principal_id       = data.databricks_service_principal.service_principal.sp_id
+  allow_cluster_create       = true
+  allow_instance_pool_create = true
+}
+```
+
+Setting entitlements to all users in a workspace - referencing special `users` [databricks_group](../data-sources/group.md)
+
+```hcl
+data "databricks_group" "users" {
+  display_name = "users"
+}
+
+resource "databricks_entitlements" "workspace-users" {
+  group_id                   = data.databricks_group.users.id
+  allow_cluster_create       = true
+  allow_instance_pool_create = true
+}
+```
+
+## Argument Reference
+
+The following arguments are available to specify the identity you need to enforce entitlements. You must specify exactly one of those arguments otherwise resource creation will fail.
+
+* `user_id` -  Canonical unique identifier for the user.
+* `group_id` - Canonical unique identifier for the group.
+* `service_principal_id` - Canonical unique identifier for the service principal.
+
+The following entitlements are available.
+
+* `allow_cluster_create` -  (Optional) Allow the user to have [cluster](cluster.md) create privileges. Defaults to false. More fine grained permissions could be assigned with [databricks_permissions](permissions.md#Cluster-usage) and `cluster_id` argument. Everyone without `allow_cluster_create` argument set, but with [permission to use](permissions.md#Cluster-Policy-usage) Cluster Policy would be able to create clusters, but within boundaries of that specific policy.
```

Comment:
isn't it kebab-case (`allow-cluster-create`)?

https://github.com/databricks/terraform-provider-databricks/blob/master/scim/scim.go#L41-L46

Reply:
we map kebab case (SCIM) to snake case (Terraform)

Reply:
ah! it's not the entitlement list. what do you think about just overriding entitlement list, instead of boolean flags, btw?

Reply:
I'm trying to keep the syntax consistent with the existing entitlements in the scim resources 👍